tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -231,32 +231,49 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
         @_rendered = []
 
     # Helper function to trigger a page rendering
+    # This is an asynchronous method; returns a promise.
     renderPage = (doc, index) ->
       if doc.isPageMapped(index)
         throw new Error "Cannot call renderPage with an already mapped index: #{index}, ensure the document is setup correctly"
 
-      doc._rendered.push index
+      new Promise (resolve, reject) ->
+        setTimeout ->
+          doc._rendered.push index
 
-      # Publish an event
-      event = document.createEvent "UIEvents"
-      event.initUIEvent "docPageMapped", false, false, window, 0
-      event.pageIndex = index
-      window.dispatchEvent event
+          # Publish an event
+          event = document.createEvent "UIEvents"
+          event.initUIEvent "docPageMapped", false, false, window, 0
+          event.pageIndex = index
+          window.dispatchEvent event
+
+          # Resolve the promise
+          resolve()
```

Comment:
Any callback passed to `then` is always resolved in a future tick so there's no need for the `setTimeout` here.